packages/api/src/opamp/controllers/opampController.ts

```diff
@@ -0,0 +1,203 @@
+    try {
+      // Check content type
+      const contentType = req.get('Content-Type');
+      if (contentType !== 'application/x-protobuf') {
+        res
+          .status(415)
+          .send(
+            'Unsupported Media Type: Content-Type must be application/x-protobuf',
+          );
+        return;
+      }
+
+      // Decode the AgentToServer message
+      const agentToServer = decodeAgentToServer(req.body);
+      logger.debug('agentToServer', agentToServer);
+      logger.debug(
+        // @ts-ignore
+        `Received message from agent: ${agentToServer.instanceUid?.toString(
+          'hex',
+        )}`,
+      );
+
+      // Process the agent status
+      const agent = agentService.processAgentStatus(agentToServer);
+
+      // Prepare the response
+      const serverToAgent: any = {
+        instanceUid: agent.instanceUid,
+        capabilities: serverCapabilities,
+      };
+
+      // Check if we should send a remote configuration
+      if (agentService.agentAcceptsRemoteConfig(agent)) {
+        const team = await getTeam();
+        // This is later merged with otel-collector/config.yaml
+        // we need to instantiate a valid config so the collector
+        // can at least start up
+        const NOP_CONFIG: CollectorConfig = {
+          extensions: {},
+          receivers: {
+            nop: null,
+          },
+          service: {
+            extensions: [],
+            pipelines: {
+              traces: {
+                receivers: ['nop'],
+              },
+              metrics: {
+                receivers: ['nop'],
+              },
+              'logs/in': {
+                receivers: ['nop'],
+              },
+            },
+          },
+        };
+        let config = NOP_CONFIG;
+
+        // If team is not found, don't send a remoteConfig, we aren't ready
+        // to collect telemetry yet
+        if (team) {
+          config = {
+            extensions: {},
+            receivers: {
+              'otlp/hyperdx': {
+                protocols: {
+                  grpc: {
+                    endpoint: '0.0.0.0:4317',
+                    include_metadata: true,
+                  },
+                  http: {
+                    endpoint: '0.0.0.0:4318',
+                    cors: {
+                      allowed_origins: ['*'],
+                      allowed_headers: ['*'],
+                    },
+                    include_metadata: true,
+                  },
+                },
+              },
+            },
+            service: {
+              extensions: [],
+              pipelines: {
+                traces: {
+                  receivers: ['otlp/hyperdx'],
+                },
+                metrics: {
+                  receivers: ['otlp/hyperdx', 'prometheus'],
+                },
+                'logs/in': {
+                  receivers: ['otlp/hyperdx', 'fluentforward'],
+                },
```

Comment:
I assume these two receivers would be authed as well?

Reply:
prometheus is internal right? fluentforward we need to in a follow ticket and likely in the pipeline like we do in v1 since there's no real auth we can do outside of that, but lmk if you have other thoughts